quantum/split_common/transactions.c

```diff
@@ -0,0 +1,670 @@
+// Backlight
+
+#ifdef BACKLIGHT_ENABLE
+
+static bool backlight_handlers_master(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) {
+    static uint32_t last_update = 0;
+    uint8_t         level       = is_backlight_enabled() ? get_backlight_level() : 0;
+    return send_if_condition(PUT_BACKLIGHT, &last_update, (level != split_shmem->backlight_level), &level, sizeof(level));
+}
+
+static void backlight_handlers_slave(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) { backlight_set(split_shmem->backlight_level); }
+
+#    define TRANSACTIONS_BACKLIGHT_MASTER() TRANSACTION_HANDLER_MASTER(backlight_handlers)
+#    define TRANSACTIONS_BACKLIGHT_SLAVE() TRANSACTION_HANDLER_SLAVE(backlight_handlers)
+#    define TRANSACTIONS_BACKLIGHT_REGISTRATIONS [PUT_BACKLIGHT] = trans_initiator2target_initializer(backlight_level),
+
+#else  // BACKLIGHT_ENABLE
+
+#    define TRANSACTIONS_BACKLIGHT_MASTER()
+#    define TRANSACTIONS_BACKLIGHT_SLAVE()
+#    define TRANSACTIONS_BACKLIGHT_REGISTRATIONS
+
+#endif  // BACKLIGHT_ENABLE
+
+////////////////////////////////////////////////////
+// RGBLIGHT
+
+#if defined(RGBLIGHT_ENABLE) && defined(RGBLIGHT_SPLIT)
+
+static bool rgblight_handlers_master(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) {
+    static uint32_t     last_update = 0;
+    rgblight_syncinfo_t rgblight_sync;
+    rgblight_get_syncinfo(&rgblight_sync);
+    if (send_if_condition(PUT_RGBLIGHT, &last_update, (rgblight_sync.status.change_flags != 0), &rgblight_sync, sizeof(rgblight_sync))) {
+        rgblight_clear_change_flags();
+    } else {
+        return false;
+    }
+    return true;
+}
+
+static void rgblight_handlers_slave(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) {
+    // Update the RGB with the new data
+    if (split_shmem->rgblight_sync.status.change_flags != 0) {
+        rgblight_update_sync(&split_shmem->rgblight_sync, false);
+        split_shmem->rgblight_sync.status.change_flags = 0;
+    }
+}
+
+#    define TRANSACTIONS_RGBLIGHT_MASTER() TRANSACTION_HANDLER_MASTER(rgblight_handlers)
+#    define TRANSACTIONS_RGBLIGHT_SLAVE() TRANSACTION_HANDLER_SLAVE(rgblight_handlers)
+#    define TRANSACTIONS_RGBLIGHT_REGISTRATIONS [PUT_RGBLIGHT] = trans_initiator2target_initializer(rgblight_sync),
+
+#else  // defined(RGBLIGHT_ENABLE) && defined(RGBLIGHT_SPLIT)
+
+#    define TRANSACTIONS_RGBLIGHT_MASTER()
+#    define TRANSACTIONS_RGBLIGHT_SLAVE()
+#    define TRANSACTIONS_RGBLIGHT_REGISTRATIONS
+
+#endif  // defined(RGBLIGHT_ENABLE) && defined(RGBLIGHT_SPLIT)
+
+////////////////////////////////////////////////////
+// LED Matrix
+
+#if defined(LED_MATRIX_ENABLE) && defined(LED_MATRIX_SPLIT)
+
+static bool led_matrix_handlers_master(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) {
+    static uint32_t   last_update = 0;
+    led_matrix_sync_t led_matrix_sync;
+    memcpy(&led_matrix_sync.led_matrix, &led_matrix_config, sizeof(led_eeconfig_t));
```

Comment:
The LED Matrix config struct is (for some reason) called `led_matrix_eeconfig`. This line doesn't compile.

Reply:
This is because the type name (`rgb_config_t` in RGB Matrix) would conflict with the `led_config_t` that `g_led_config` uses. In fact, `rgb_config_t` should really be renamed to `rgb_eeconfig_t` to match, but I think there are some references to it in user keymaps, so it is not as trivial as the `point_t` rename.

Reply:
Makes sense, but I was referring to the variable name, not the type. ;)

Reply:
The variable name is a result of me having to break from the naming convention, or rather fix it, for LED Matrix. It would also be renamed in the RGB Matrix code along with the type name, and I'd also do the same for eg. `backlight_config_t` -> `backlight_eeconfig_t`.

Reply:
@tzarc This line and the one I commented on below still don't compile (with `LED_MATRIX_ENABLE`). The global LED matrix config variable is named `led_matrix_eeconfig` (not `led_matrix_config`).

Reply:
Serves me right for copypasting rgb_matrix code.

Reply:
> 
> 
> Serves me right for copypasting rgb_matrix code.

That sounds like fighting words =p